File: intro-langchain-typescript/src/examples/1-fundamentos/1-hello_world.ts
import { ChatOpenAI } from "@langchain/openai";
import {
  loadEnvironment,
  getApiKey,
  getEndpoint,
  displayResult,
} from "../../utils/helpers";

// Carrega as variáveis de ambiente
loadEnvironment();

async function main(): Promise<void> {
  try {
    // Configura o modelo ChatOpenAI
    const model = new ChatOpenAI({
      model: "gpt-4o",
      temperature: 0.5,
      maxTokens: 100,
      configuration: {
        baseURL: getEndpoint("GITHUB_MODELS_ENDPOINT"),
        apiKey: getApiKey("GITHUB_MODELS_TOKEN"),
      },
    });

    // Faz a chamada para o modelo
    const response = await model.invoke("Hello World!");

    // Exibe o resultado
    displayResult("LangChain Hello World (TypeScript)", response.content);
  } catch (error) {
    console.error("Erro:", error instanceof Error ? error.message : error);
  }
}

// Executa a função principal
main();

File: intro-langchain-python/examples/1-fundamentos/1-hello_world.py
from langchain_openai import ChatOpenAI
from dotenv import load_dotenv
from pydantic import SecretStr
import os
load_dotenv()

model = ChatOpenAI(
    model="gpt-4o", 
    temperature=0.5,
    max_completion_tokens=100,
    base_url=os.environ["GITHUB_MODELS_ENDPOINT"],
    api_key=SecretStr(os.environ["GITHUB_MODELS_TOKEN"])
    )

message = model.invoke("Hello World!")
print(message.content)